.github/workflows/pr_preview.yml

```diff
@@ -0,0 +1,64 @@
+name: Surge PR Preview
+
+on:
+  pull_request:
+    # when using teardown: 'true', add default event types + closed event type
+    types: [opened, synchronize, reopened, closed]
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Checkout RIOT
+        uses: actions/checkout@v2
+        with:
+          repository: 'RIOT-OS/RIOT'
+          path: _RIOT
+          fetch-depth: 0
+      - name: Fetch boards
```

Comment:
all these steps are replicated from the deployment job (deployment.yml). this replication is error-prone when changes occur. it would be better to re-use the definition of steps, i.e., define the steps once and use them multiple times.

Reply:
That would require external tooling. But that wouldn't be a bad thing as with that, a developer could also fetch the boards etc locally.